### PR TITLE
Add uint support into the write protocol

### PR DESCRIPTION
**Pull request by jsternberg**
- [ ] Rebased/mergable
- [ ] Tests pass
- [ ] CHANGELOG.md updated
- [ ] Provide example syntax
- [ ] [InfluxData Documentation](https://github.com/influxdata/docs.influxdata.com): issue filed or pull request submitted \<link to issue or pull request\>

### CODE REVIEW

File: models/points.go [hunk 1]
```diff
@@ -926,6 +933,22 @@ func scanNumber(buf []byte, i int) (int, error) {
 				return i, fmt.Errorf("unable to parse integer %s: %s", buf[start:i-1], err)
 			}
 		}
+	} else if isUnsigned {
+		// Make sure the last char is a 'u' for unsigned
+		if buf[i-1] != 'u' {
+			return i, ErrInvalidNumber
+		}
+		// Make sure the first char is not a '-' for unsigned
```
Comment (rbetts) [inline models/points.go]: This check is redundant with the initial check at line 833?

File: models/points_test.go [hunk 2]
```diff
@@ -2173,7 +2175,7 @@ m v=42i
 m v="string"
 m v=true
 m v="string\"with\"escapes"
-m v=42i,f=42,g=42.314
```
Comment (rbetts) [inline models/points_test.go]: I might be reading the test incompletely - but need to be sure we have tests for a negative unsigned int (error path), for max value, and for something greater than the max value.

File: models/points.go (hunk 1 above)
Comment (jsternberg) [reply models/points.go]: I don't believe so. The check at line 833 prevents a person from writing `m=-`. This checks to make sure a negative wasn't used while looking at an unsigned integer.

File: models/points_test.go (hunk 2 above)
Comment (jsternberg) [reply models/points_test.go]: I'll add those.

**Comment by jsternberg** [comment]
I wasn't planning on having this merged until the query engine stuff was done, but I didn't think about the other applications. Should I build it as a configuration option or some compilation switch? I can also just block all incoming writes of unsigned with no compilation switch and no configuration option.

**Comment by jsternberg** [comment]
It also may be difficult to have a switch that enables this. The section of code that does the parsing doesn't really have access to any configuration and catching it later than that would probably add some undesired overhead.

**Comment by phemmer** [comment]
also is there any planned way to be able to change the type of an existing field?
I'm just imagining the scenario in telegraf where the various input plugins are storing their types as `uint64`, and the telegraf marshaller is sending them as signed integers. But once this PR is merged, and the telegraf marshaller is changed so that `uint64` is actually sent as an unsigned integer, lots of writes will start failing because of conflicting field type.

**Comment by rbetts** [comment]
@phemmer The ability to retroactively rewrite field types is not in scope. We are thinking through issues like the one you've raised but our immediate goal is to get to a functional uint64 field type.

**Comment by rbetts** [comment]
@jsternberg I would avoid a configuration parameter - as this won't be long lived toggle once everything is done.  A command line flag or environment variable might be preferable. I would like to get these bits landed in reasonable chunks on master.

**Comment by jsternberg** [comment]
@rbetts done.

```
$ go install ./cmd/influxd
$ influx
> create database db0
> use db0
> insert cpu value=1u
ERR: {"error":"unable to parse 'cpu value=1u': invalid number"}

> exit
$ go install -tags uint ./cmd/influxd
$ influx
> use db0
> insert cpu value=1u
> exit
```

**Comment by jsternberg** [comment]
I need to fix the tests since they fail since uint support is turned off. I'm also adding some additional parser tests for uint64 on top of the existing ones.
